### ISSUE TITLE
CBVCorrector: Kepler and K2 CBV time stamps are not barycenter corrected

### ISSUE THREAD

**Issue by jcsmithhere** [lightkurve/lightkurve#965]
The Kepler and K2 CBV time stamps on MAST are not barycenter corrected. Whereas the POC light curves on MAST *are* barycenter corrected. This causes a minor issue where the light curves and the CBVs are not perfectly aligned in CBVCorrector. The TDB correction is only about 8 minutes maximum for Kepler so the error is not huge compared to the 30 minute long cadence. Nonetheless, the CBV time stamps should be converted to TDB when loaded into CBVCorrector.

**Comment by barentsen** [comment]
Thanks for calling attention to this Jeff! :+1:

I don't see this fact mentioned in Section 2.3.4 of the [Kepler Archive Manual](https://archive.stsci.edu/kepler/manuals/archive_manual.pdf).  It may be good to include this knowledge in the `KeplerCotrendingBasisVectors` class.  I imagine it could be mentioned as a caveat in the docstring, and the class could perhaps offer a correction method?

Are you confident that the TESS CBVs have the barycentric correction applied?  This would surprise me because the exact barycentric correction is a function of an object's (ra, dec) position, whereas the CBV vectors cover an entire CCD.

The FITS headers offer the following opinion:

### Time unit in a Kepler light curve file
```python
>>> from astropy.io import fits
>>> f = fits.open("https://archive.stsci.edu/missions/kepler/lightcurves/0080/008004724/kplr008004724-2009259160929_llc.fits")
>>> f[1].header['TUNIT1']
'BJD - 2454833'  # barycentric
```
### Time unit in a Kepler CBV file
```python
>>> f = fits.open("https://archive.stsci.edu/missions/kepler/cbv/kplr2010265121752-q06-d09_lcbv.fits")
>>> f[1].header['TUNIT1']
'MJD'  # not barycentric
```
### Time unit in a TESS light curve file
```python
>> f = fits.open("https://mast.stsci.edu/api/v0.1/Download/file/?uri=mast:TESS/product/tess2018206045859-s0001-0000000220578510-0120-s_lc.fits")
>>> f[1].header['TUNIT1']
'BJD - 2457000, days'   # barycentric
```

### Time unit in a TESS CBV file
```python
>>> f = fits.open("https://archive.stsci.edu/missions/tess/ffi/s0033/2020/351/1-4/tess2020351194500-s0033-1-4-0203-s_cbv.fits")
>>> f[1].header['TUNIT1']
'JD - 2457000, days'  # not barycentric?
```

**Comment by jcsmithhere** [comment]
Yes, I should make a note in the CBVCorrector classes and the tutorial about this. The TESS CBV FITS file headers contains lines indicating they are indeed Barycenter corrected:

```
TIMEREF = 'SOLARSYSTEM'        / barycentric correction applied to times
TASSIGN = 'SPACECRAFT'         / where time is assigned
TIMESYS = 'TDB     '           / time system is Barycentric Dynamical Time (TDB)
```

I also recall people saying during mission planning that they will be barycenter corrected. You make a good point about the correction varies across the very large TESS FOV. I assume the correction is at the center of each CCD. Perhaps the error is not too large over just a single CCD?

In any event, the systematics are local to the spacecraft, so what we really should do is reverse the barycenter correction for each target to place it in spacecraft time, then perform the CBV correction. (PDC performs the correction in local spacecraft time, which is why I never had to deal with this before.) Do you know how easy it is tp perform the reverse barycenter correction in astropy?

**Comment by barentsen** [comment]
Putting a light curve back into non-barycentric is easy for official light curves thanks to the TIMECORR column and TIMSLICE header value in the light curve files.  The relationship is documented on page 17 in the archive manual:

<img width="885" alt="Screen Shot 2021-02-22 at 12 35 38 PM" src="https://user-images.githubusercontent.com/817669/108767032-bc0ca480-750a-11eb-8091-8a7fb02482e3.png">

For user-created light curves, we'd have to propagate this information from the target pixel file.  This will not be easy for light curves created from TESSCut files, so a better solution may be to do CBV alignment via cadence numbers instead of times?

**Comment by jcsmithhere** [comment]
Yes, alignment via cadence numbers is safer and it's the default. The problem is I'm not sure if we can reliably interpolate using cadence numbers. I think cadence numbers advance at the cadence time length, even during non-science collection times but I'm not sure if that can be relied upon, hence I use time stamps to interpolate. If cadence length does not change and the cadence number always advances reliably then perhaps we can switch to using that for interpolation.

I think we really should reverse the barycenter correction and perform the CBV correction in spacecraft time.  This is more an issue with TESS with 2- minute cadence, and especially important for the 20-second cadence data!
